From 5a57792ca459b14cfdd14bd38bf520c3bc60fffc Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 02 Apr 2020 00:55:24 +0300
Subject: [PATCH] [SW-11180] preboot: move u-boot offset assembly

Move the U-Boot offset assembly to next_image_offset() function solely.
This will enable to (re)use the function with the new boot descriptors.

Change-Id: I73650305f8d3fd5670cfc4662e7db811e1b7eed7
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/preboot.c b/app/preboot.c
index dc88745..b759c59 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -68,21 +68,22 @@
 {
 	u32 upd_pending_reg = get_upd_pending_sts_reg();
 	u32 image_pending = regmap_readl(upd_pending_reg);
+	u32 base_off;
 
 	/* Check for normal boot */
 	if (image_pending != FLOPS_UPD_PENDING)
-		return stable_image_base_offset();
+		base_off = stable_image_base_offset();
+	else /* There is an update pending */
+		base_off = modifiable_image_base_offset(pending);
 
-	/* There is an update pending */
-	return modifiable_image_base_offset(pending);
+	return base_off + HL_SPI_LEGACY_UBOOT_OFF;
 }
 
 static void preboot_copy_uboot(void)
 {
 	char *name = "U-Boot";
 	bool upgrade_pending = false;
-	u32 img_offset = next_image_offset(&upgrade_pending);
-	u32 uboot_offset = img_offset + HL_SPI_LEGACY_UBOOT_OFF;
+	u32 uboot_offset = next_image_offset(&upgrade_pending);
 	char *uboot_image = upgrade_pending ? "candidate" : "stable";
 
 	preboot_puts_raw("Preparing to run U-Boot ");
